todoapp/app/src/main/java/com/example/android/architecture/blueprints/todoapp/taskdetail/domain/logic.kt

```diff
@@ -0,0 +1,38 @@
+@file:JvmName("TaskDetailLogic")
+
+package com.example.android.architecture.blueprints.todoapp.taskdetail.domain
+
+import com.example.android.architecture.blueprints.todoapp.data.Task
+import com.spotify.mobius.Effects.effects
+import com.spotify.mobius.Next
+import com.spotify.mobius.Next.*
+
+fun update(task: Task, event: TaskDetailEvent): Next<Task, TaskDetailEffect> =
+        when (event) {
+            TaskDetailEvent.DeleteTaskRequested -> dispatch(effects(TaskDetailEffect.DeleteTask(task)))
```

Comment:
This probably won't be valid if you change the cases so they're not inner classes anymore. If you don't though, can we at least statically import them? `TaskDetailEvent` and `TaskDetailEffect` here seem very noisy.